### PR TITLE
fix: require postcss module via resolver

### PR DESCRIPTION
**Pull request by TheAlexLichter**
Can be useful when trying to run a nuxt app in another folder, eg. via

`yarn nuxt ../my-app` or `yarn nuxt -c ../my-app/nuxt.config.js`

Otherwise, PostCSS plugins will be taken from the current working dir which probably doesn't have the plugins (in the above case it's the root dir of the nuxtjs repo)

## Types of changes
- [x] Bug fix (a non-breaking change which fixes an issue)
- [ ] New feature (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly. (PR: #)
- [ ] I have added tests to cover my changes (if not applicable, please state why)
- [x] All new and existing tests are passing.

### CODE REVIEW

File: packages/webpack/src/utils/postcss.js [hunk 1]
```diff
@@ -6,6 +6,7 @@ import cloneDeep from 'lodash/cloneDeep'
 import createResolver from 'postcss-import-resolver'
 
 import { isPureObject } from '@nuxt/utils'
+import { requireModule } from '@nuxt/cli/src/utils'
```
Comment (pi0) [inline packages/webpack/src/utils/postcss.js]: By design, No non-distro package should depend on `@nuxt/cli`

File: packages/webpack/src/utils/postcss.js [hunk 2]
```diff
@@ -113,7 +114,7 @@ export default class PostcssConfig {
       // Map postcss plugins into instances on object mode once
       config.plugins = this.sortPlugins(config)
         .map((p) => {
-          const plugin = require(p)
+          const plugin = requireModule(this.nuxt.resolver.resolveModule(p))
```
Comment (pi0) [inline packages/webpack/src/utils/postcss.js]: Can we use `this.nuxt.resolver.requreModule(p)` ? :)

File: packages/webpack/src/utils/postcss.js (hunk 1 above)
Comment (manniL) [reply packages/webpack/src/utils/postcss.js]: Whoops. That was not intended 🙈

**Review by pi0 — DISMISSED** [review]

File: packages/webpack/src/utils/postcss.js [hunk 3]
```diff
@@ -113,7 +113,7 @@ export default class PostcssConfig {
       // Map postcss plugins into instances on object mode once
       config.plugins = this.sortPlugins(config)
         .map((p) => {
-          const plugin = require(p)
+          const plugin = this.nuxt.resolver.requireModule(this.nuxt.resolver.resolveModule(p))
```
Comment (clarkdo) [inline packages/webpack/src/utils/postcss.js]: What's the purpose of require twice ?

Comment (manniL) [reply packages/webpack/src/utils/postcss.js]: It's not required twice.

`resolveModule` will get you the correct path (for the project), eg. `/path/to/project/node_modules/tailwindcss/lib/index.js`

`requireModule` will then require the resolved path.

If you mean that we could switch out `requireModule` for `require`, yes, that'd be possible

Comment (clarkdo) [reply packages/webpack/src/utils/postcss.js]: I mean that it seemed `requireModule` will resolve the path as well 😃

Comment (pi0) [reply packages/webpack/src/utils/postcss.js]: I agree. requireModule internally does it itself

Comment (manniL) [reply packages/webpack/src/utils/postcss.js]: Ohh! Good point 👍

**Review by pi0 — APPROVED** [review]

**Pull request merged**